pkg/arvo/app/metadata-hook.hoon

```diff
@@ -0,0 +1,200 @@
+  |^
+  ?-  -.act
+      %add-owned
+    ?>  (team:title our.bowl src.bowl)
+    :-  ~
+    ?:  (~(has by synced) path.act)  state
+    state(synced (~(put by synced) path.act our.bowl))
+  ::
+      %add-synced
+    ?>  (team:title our.bowl src.bowl)
+    =/  =path  [%group path.act]
+    ?:  (~(has by synced) path.act)  [~ state]
+    :_  state(synced (~(put by synced) path.act ship.act))
+    [%pass path %agent [ship.act %metadata-hook] %watch path]~
+  ::
+      %remove
+    =/  ship  (~(get by synced) path.act)
+    ?~  ship  [~ state]
+    ?:  &(?!(=(u.ship our.bowl)) ?!((team:title our.bowl src.bowl)))
+      [~ state]
+    :_  state(synced (~(del by synced) path.act))
+    %-  zing
+    :~  (unsubscribe [%group path.act] u.ship)
+        ?.  &(=(u.ship our.bowl) (team:title our.bowl src.bowl))  ~
+        [%give %kick ~[[%group path.act]] ~]~
+    ==
+  ==
+  ::
+  ++  unsubscribe
+    |=  [=path =ship]
+    ^-  (list card)
+    ?:  =(ship our.bowl)
+      [%pass path %agent [our.bowl %metadata-store] %leave ~]~
+    [%pass path %agent [ship %metadata-hook] %leave ~]~
+  --
+::
+++  watch-group
+  |=  =path
+  ^-  (list card)
+  |^
+  ?>  (~(has by synced) path)
+  ?>  (is-permitted [(scot %p src.bowl) path])
+  %+  turn  ~(tap by (metadata-scry path))
+  |=  [[=group-path =resource] =metadata]
+  ^-  card
+  [%give %fact ~ %metadata-update !>([%add group-path resource metadata])]
+  ::
+  ++  is-permitted
+    |=  pax=^path
+    ^-  ?
+    =.  pax
+      ;:(weld /=permission-store/(scot %da now.bowl)/permitted pax /noun)
+    .^(? %gx pax)
+  ::
+  ++  metadata-scry
+    |=  pax=^path
+    ^-  associations
+    =.  pax  ;:(weld /=metadata-store/(scot %da now.bowl)/group pax /noun)
+    (need .^((unit associations) %gx pax))
+  --
+::
+++  fact-metadata-update
+  |=  [wir=wire fact=metadata-update]
+  ^-  (quip card _state)
+  |^
+  [?:((team:title our.bowl src.bowl) handle-local handle-foreign) state]
+  ::
+  ++  handle-local
+    ?+  -.fact  ~
+        %add
+      ?.  (~(has by synced) group-path.fact)  ~
+      (give group-path.fact fact)
+    ::
+        %update-metadata
+      ?.  (~(has by synced) group-path.fact)  ~
+      (give group-path.fact fact)
+    ::
+        %remove
+      ?.  (~(has by synced) group-path.fact)  ~
+      (give group-path.fact fact)
+    ==
+  ::
+  ++  handle-foreign
+    ?+  -.fact  ~
+        %add
+      ?.  =(src.bowl (~(got by synced) group-path.fact))  ~
+      (poke fact)
+    ::
+        %update-metadata
+      ?.  =(src.bowl (~(got by synced) group-path.fact))  ~
+      (poke [%add +.fact])
+    ::
+        %remove
+      ?.  =(src.bowl (~(got by synced) group-path.fact))  ~
+      (poke fact)
+    ==
+  ::
+  ++  give
+    |=  [=path upd=metadata-update]
+    ^-  (list card)
+    [%give %fact ~[[%group path]] %metadata-update !>(upd)]~
+  ::
+  ++  poke
+    |=  act=metadata-action
+    ^-  (list card)
+    [%pass / %agent [our.bowl %metadata-store] %poke %metadata-action !>(act)]~
```

Comment:
`+forward` and `+store` might be more appropriate here. These aren't generic "build a card" arms.

Reply:
I prefer give and poke, as they're literal. Given that they're already scoped to this particular arm, I see no issue with the shorthand names.

Reply:
pass/give would be parallel, as would poke/fact, or update/action.